hat/app/org/hatdex/hat/api/controllers/Authentication.scala

```diff
@@ -618,10 +604,101 @@ class Authentication @Inject() (
+    * This is unauthenticated request
+    */
+  def handleRequestVerification: Action[ApiValidationRequest] =
+    UserAwareAction.async(parsers.json[ApiValidationRequest]) {
+      implicit request =>
+        logger.debug("Processing resend validation request")
+
+        val email = request.body.email
+        val applicationId = request.body.applicationId
+        val isSigupToken = true
+
+        // Generic message regardless if the user is found or not.
+        val uniformResponse = resendVerificationEmail
+
+        // The email from the request and the Silhouette Env do not match
+        if (email != request.dynamicEnvironment.ownerEmail) {
+          Future.successful(uniformResponse)
+        }
+        else {
+          val enventuallTtokenToSend = for {
+            token <- tokenService.retrieve(email, isSigupToken)
+            tokenToSend <- createOrResendToken(email, token)
+          } yield tokenToSend
+
+          enventuallTtokenToSend.flatMap { tokenOpt =>
+            tokenOpt match {
+              case Some(token) => {
+                // Is Owner and is NOT EmailVerified
+                val eventuallyUsers = usersService.listUsers
+                eventuallyUsers.flatMap { users => 
+                  users.find(u => (u.roles.contains(Owner()) && !(u.roles.contains(EmailVerified)))) match {
+                    case Some(user) => {
+                      sendRevalidationEmail(user.email, applicationId, request.host, token, request.lang)
+                      Future.successful(uniformResponse)
+                    }
+                    // The user is not an owner, but return the "If we found an email address, we'll send the link."
+                    case None => {
+                      Future.successful(uniformResponse)
+                    }
+                  }
+                } 
+              }
+              case None => {
+                // User does not have an existing isSignup Token
+                Future.successful(uniformResponse)
+              }
+            }
+          }
+            
+          Future.successful(uniformResponse)
+        }
+      }
+
+  // applicationId is currently unused
+  def sendRevalidationEmail(email: String, applicationId: String, requestHost: String, token: MailTokenUser, lang: Lang)(implicit hatServer: HatServer): Future[Done] = { 
+    implicit val l = lang
+    tokenService.create(token).map { _ =>
+      // Assume https now
+      val claimLink = s"https://${requestHost}/hat/claim/${token.id}?email=${URLEncoder.encode(email, "UTF-8")}"
+      mailer.claimHat(email, claimLink, None)
+    }
+  }
+
+  def createOrResendToken(email: String, tokenOpt: Option[MailTokenUser])(implicit hatServer: HatServer): Future[Option[MailTokenUser]] = {
+    tokenOpt match {
+      case Some(token) => {
+        token.isExpired match {
+          // If the token has expired, delete it, create a new one, send that in the email
+          case true => {
```

Comment:
Braces for cases are redundant, IntelliJ is complains about them. Although I think Scalafmt has removed them upon file save.